### ISSUE TITLE
Max Instance more than 50 runs very slowly and delays execution time

### ISSUE THREAD

**Issue by jagannath13** [webdriverio/webdriverio#2380]
I have 200 concurrent VM's via Sauce to execute, if i run 300 test on 200 threads Performance is very slow and CPU goes crazy, i still have 8GB RAM in Local, tried with Jenkins and Concourse plugins as well,

50 threads run smoothly and test gets completed less than 20 mins
let me know if any further options are config need to be added if max instance is 200

what is the suggested max thread for WDIO

**Comment by christian-bromann** [comment]
> what is the suggested max thread for WDIO

This depends on the CPU of your machine. We are trying to run tests as efficient as possible but at a certain point the CPU will be fully utilised.

**Comment by jagannath13** [comment]
thanks Chris In our previous JAVA based frameworks it used to run fine 200 in parallel, currently this is the only bigger issue for us implement across Organization

**Comment by monolithed** [comment]
The problem due to process forks. I guess that forks should be replaced with threads/fibers and the most performance problems will be solved

**Comment by christian-bromann** [comment]
Closing in favor of #1643
